doc/rtd/reference/cli.rst

```diff
@@ -487,4 +498,52 @@ Which would produce the following example output:
       "status": "done"
     }
 
+cloud-id
+========
+
+``cloud-id`` reports the canonical cloud-id for the instance.
```

Comment:
All valid questions. I wasn't aware that we had a separate mapping and I have the same questions for that.

For your first two questions, I don't think I would want to document each cloud-id, and I'm not sure "figure yours out if you need to use it" is too big an ask.

Is there something you think should be addressed here?

Reply:
> I don't think I would want to document each cloud-id

Another column in [this table](https://docs.cloud-init.io/en/latest/reference/datasource_dsname_map.html#datasource-dsname) page would be a valuable addition in my opinion.

> I'm not sure "figure yours out if you need to use it" is too big an ask.

If they need this for anything besides informational output, they run instances on different clouds and want to differentiate between them. If a user whose code was tested to work in one region using:

```bash
if [ cloud-id = "aws" ]; then
    # do thing
else
    # do different thing
fi
```

starts doing the wrong thing due to `cloud-id` returning `aws-indonesia` or something like that, they might be understandably confused given the current docs.

> Is there something you think should be addressed here?

Think we should file an issue to collect for a column in the dsname table? 

There is a part of me that wants to leave it undocumented and forgotten, but since we already ship it we should probably document it. At a minimum, lets provide a cautionary statement in the documentation so that users don't expect it to behave reliably since our confidence that it will do that is apparently relatively low (given the unanswered questions above). Without much confidence in its output it is hard to know what we even can say about it that is true besides "if you run it on one instance in one region, it will give you the same output again on a different instance in the same region, probably". Even the examples added here might not be correct in all cases.

Reply:
> so that users don't expect it to behave reliably since our confidence that it will do that is apparently relatively low 

AFAIK, the Pro client already has a fairly strong reliance on it.

@blackboxsw , do you have any context here since you're the original author and you might better understand Pro's use cases? The map that Brett points to does seem very small and I'm curious why so few regions are represented. Why are we calling out those three regions separately?